src/main/java/it/pagopa/transactions/utils/UpdateTransactionStatusTracerUtils.java

```diff
@@ -0,0 +1,152 @@
+package it.pagopa.transactions.utils;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * This utility class traces transaction update status performed by external
+ * entities. Tracing is performed by meaning of OpenTelemetry span creation.
+ * {@link UpdateTransactionStatusTracerUtils.UpdateTransactionStatusType}
+ * enumeration contains transaction status typologies enumeration
+ * {@link UpdateTransactionStatusTracerUtils.UpdateTransactionTrigger}
+ * enumeration, instead, contains external actors that trigger the transaction
+ * status update
+ */
+@Component
+public class UpdateTransactionStatusTracerUtils {
+
+    private final OpenTelemetryUtils openTelemetryUtils;
+
+    /**
+     * Span attribute used to discriminate transaction update status operation type
+     *
+     * @see UpdateTransactionStatusType
+     */
+    static final AttributeKey<String> UPDATE_TRANSACTION_STATUS_TYPE_ATTRIBUTE_KEY = AttributeKey
+            .stringKey("updateTransactionStatus.type");
+    /**
+     * Spqn attribute used to trace status update outcome
+     *
+     * @see UpdateTransactionStatusOutcome
+     */
+    static final AttributeKey<String> UPDATE_TRANSACTION_STATUS_OUTCOME_ATTRIBUTE_KEY = AttributeKey
+            .stringKey("updateTransactionStatus.outcome");
+    /**
+     * Spqn attribute used to trace status update trigger
+     *
+     * @see UpdateTransactionTrigger
+     */
+    static final AttributeKey<String> UPDATE_TRANSACTION_STATUS_TRIGGER_ATTRIBUTE_KEY = AttributeKey
+            .stringKey("updateTransactionStatus.trigger");
+
+    /**
+     * Enumeration of all operation that update transaction status performed by
+     * external entities
+     */
+    public enum UpdateTransactionStatusType {
+        /**
+         * Transaction status update operation triggered by payment gateway by receiving
+         * authorization outcome
+         */
+        AUTHORIZATION_OUTCOME,
+        /**
+         * Transaction status update operation triggered by Nodo with sendPaymentResult
+         * operation
+         */
+        SEND_PAYMENT_RESULT_OUTCOME
+    }
+
+    /**
+     * Enumeration of all actors that can trigger a transaction status update
+     */
+    public enum UpdateTransactionTrigger {
+        /**
+         * Transaction status update triggered by Nodo
+         */
+        NODO,
+        /**
+         * Transaction status update triggered by NPG (authorization outcome)
+         */
+        NPG,
+        /**
+         * Transaction status update triggered by PGS (VPOS)
+         */
+        PGS_VPOS,
+
+        /**
+         * Transaction status update triggered by PGS (XPAY)
+         */
+        PGS_XPAY
+    }
```

Comment:
I'm not convinced by this encoding, as it allows invalid combinations (such as `(NPG, SEND_PAYMENT_OUTCOME)`)

Reply:
Made span parameter combination encoded into code in order to not allow for invalid combination.
See [this commit](https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/390/commits/360e43bfd75d3c71feb9058497859071365853f5)